starlite/testing/life_span_handler.py

```diff
@@ -1,28 +1,29 @@
 from math import inf
-from typing import TYPE_CHECKING, Optional, Union, cast
+from typing import TYPE_CHECKING, Generic, Optional, TypeVar, cast
 
 from anyio import create_memory_object_stream
 from anyio.streams.stapled import StapledObjectStream
 
 if TYPE_CHECKING:
-    from starlite.testing import AsyncTestClient, TestClient
     from starlite.types import LifeSpanReceiveMessage  # noqa: F401  # nopycln: import
     from starlite.types import (
         LifeSpanSendMessage,
         LifeSpanShutdownEvent,
         LifeSpanStartupEvent,
     )
 
+T = TypeVar("T")
```

Comment:
T is not bound to any type, it therefore is not typed.

```suggestion
T = TypeVar("T", bound=Union[AsyncTestClient, TestClient])
```